src/server/create/prototype/controllers/create.js

```diff
@@ -0,0 +1,87 @@
+import { config } from '~/src/config/config.js'
+import { sessionNames } from '~/src/server/common/constants/session-names.js'
+import { provideCreate } from '~/src/server/create/helpers/pre/provide-create.js'
+import { buildErrorDetails } from '~/src/server/common/helpers/build-error-details.js'
+import { provideAuthedUser } from '~/src/server/common/helpers/auth/pre/provide-authed-user.js'
+import { auditMessageCreated } from '~/src/server/common/helpers/audit/messages/audit-message-created.js'
+import { scopes } from '~/src/server/common/constants/scopes.js'
+import { prototypeValidation } from '~/src/server/create/prototype/schema/prototype-validation.js'
+
+const prototypeCreateController = {
+  options: {
+    auth: {
+      mode: 'required',
+      access: {
+        scope: [scopes.admin, '{payload.teamId}']
+      }
+    },
+    pre: [provideCreate, provideAuthedUser]
+  },
+  handler: async (request, h) => {
+    const create = request.pre?.create
+    const repositoryName = create.repositoryName
+    const templateTag = create.templateTag
+    const teamId = request.payload?.teamId
+
+    const sanitisedPayload = {
+      repositoryName,
+      teamId,
+      templateTag
+    }
+
+    const validationResult = await prototypeValidation()
+      .validateAsync(sanitisedPayload, { abortEarly: false })
+      .then((value) => ({ value }))
+      .catch((error) => ({ value: sanitisedPayload, error }))
```

Comment:
so theres a mix of `await` and chanied promise work here. Not sure why? Is this working as expected?
```
const validationResult = await prototypeValidation()
      .validateAsync(sanitisedPayload, { abortEarly: false })```

should be fine by itself